src/dev/TMP117.cpp

```
@@ -0,0 +1,31 @@
+#include <core/io/I2C.hpp>
+#include <dev/TMP117.hpp>
+
+namespace TMS {
+
+TMP117::TMP117(io::I2C* i2c, uint8_t i2cSlaveAddress) : i2cSlaveAddress(i2cSlaveAddress), i2c(i2c) {}
+
+TMP117::TMP117() : i2c(nullptr) {}
+
+int16_t TMP117::readTemp() {
+    uint8_t tempBytes[2];
+    int16_t temp;
+
+    io::I2C::I2CStatus status = i2c->readReg(i2cSlaveAddress, (uint8_t*) TEMP_REG, 1, tempBytes, 2);
+
+    if (status == io::I2C::I2CStatus::OK) {
+        temp = static_cast<int16_t>(((uint16_t) tempBytes[0]) << 8 | tempBytes[1]);
+    } else {
+        temp = 0;
```

once again, do we want to silently fail? or should we do like we normally do and return the I2CStatus and simply have a ptr to a uint16_t that gets written to?